src/selectedtests/work_items/process_work_items.py

```diff
@@ -0,0 +1,128 @@
+    """
+    _clear_in_progress_work(mongo.test_mappings_queue())
+    try:
+        while True:
+            out_of_work = _process_one_work_item(evg_api, mongo, after_date, before_date)
+            if out_of_work:
+                break
+    except:  # noqa: E722
+        LOGGER.warning("Unexpected exception processing test mapping work item", exc_info=1)
+
+
+def _process_one_work_item(
+    evg_api: EvergreenApi, mongo: MongoWrapper, after_date: datetime, before_date: datetime
+) -> bool:
+    """
+    Process a test mapping work item.
+
+    :param evg_api: An instance of the evg_api client
+    :param mongo: An instance of MongoWrapper.
+    :param after_date: The date at which to start analyzing commits of the project.
+    :param before_date: The date up to which we should analyze commits of the project.
+    :return: Whether all work items have been processed.
+    """
+    log = LOGGER
+    log.info("Starting test mapping work item processing")
+    work_item = ProjectTestMappingWorkItem.next(mongo.test_mappings_queue())
+    if not work_item:
+        log.info("No more test mapping work items found")
+        return True
+
+    log = log.bind(project=work_item.project, module=work_item.module)
+    log.info("Starting test mapping work item processing for work_item")
+    if _run_create_test_mappings(evg_api, mongo, work_item, after_date, before_date, log):
+        work_item.complete(mongo.test_mappings_queue())
+
+    return False
+
+
+def _run_create_test_mappings(
+    evg_api: EvergreenApi,
+    mongo: MongoWrapper,
+    work_item: ProjectTestMappingWorkItem,
+    after_date: datetime,
+    before_date: datetime,
+    log: Any,
+) -> bool:
+    """
+    Generate task mappings for a given work item.
+
+    :param evg_api: An instance of the evg_api client
+    :param mongo: An instance of MongoWrapper.
+    :param work_item: An instance of ProjectTestMappingWorkItem.
+    :param after_date: The date at which to start analyzing commits of the project.
+    :param before_date: The date up to which we should analyze commits of the project.
+    :return: Whether test mappings were created successfully.
+    """
+    source_re = re.compile(work_item.source_file_regex)
+    test_re = re.compile(work_item.test_file_regex)
+    module_source_re = None
+    module_test_re = None
+    if work_item.module:
+        module_source_re = re.compile(work_item.module_source_file_regex)
+        module_test_re = re.compile(work_item.module_test_file_regex)
+
+    test_mappings_list = generate_test_mappings(
+        evg_api,
+        work_item.project,
+        source_re,
+        test_re,
+        after_date,
+        before_date,
+        work_item.module,
+        module_source_re,
+        module_test_re,
+    )
+    if test_mappings_list:
+        mongo.test_mappings().insert_many(test_mappings_list)
+    log.info("Finished test mapping work item processing")
+    return True
```

Comment:
[nit] Will this ever not return `True`? If not, would it be worth not having it return anything?